lib/extensions/materials/active_record/base.rb

```diff
@@ -0,0 +1,18 @@
+module Extensions::Materials::ActiveRecord::Base
+  module ClassMethods
+    # Declare this to allow models to support materials uploads.
+    def has_one_folder # rubocop:disable Style/PredicateName
+      has_one :folder, as: :owner, class_name: Course::Material::Folder.name,
+                       inverse_of: :owner, dependent: :destroy
+      has_many :materials, through: :folder, class_name: Course::Material.name
+
+      before_save :save_folder
```

Comment:
So suppose you have uploaded some materials to an assessment folder, the n we have the code like `assessment.folder.materials.build()`. But then when the assessment is saved, those materials are not saved.